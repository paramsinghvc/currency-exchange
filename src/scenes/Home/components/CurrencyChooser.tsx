import React, { FC, useCallback } from "react";
import styled from "@emotion/styled";

import Modal from "shared/components/Modal";
import theme from "shared/theme";
import { Currency } from "shared/models/Currency";
import { IReduxOperations } from "@mollycule/redux-operation";
import Anime from "shared/components/Anime";

const Holder = styled.main`
  padding: 20px;
  height: calc(100% - 79px);
  overflow-y: auto;
`;

const ListItem = styled.section`
  width: 100%;
  display: flex;
  flex-direction: column;
  margin-bottom: 30px;
  cursor: pointer;
`;

const CurrencyCode = styled.p`
  font-size: 24px;
  color: ${theme.primaryColor};
`;

const CurrencySubText = styled.p`
  font-size: 24px;
  color: ${theme.subTextColor};
  font-size: 9px;
  font-weight: 500;
  margin-top: 7px;
`;

const CurrencyChooser: FC<{
  open: boolean;
  onClose: () => void;
  currencyData: IReduxOperations<Currency[]>;
  onChange: (val: Currency) => void;
}> = ({ open, onClose, currencyData, onChange }) => {
  const handleClick = useCallback(
    (currency: Currency) => () => {
      onChange(currency);
      onClose();
    },
    [onChange, onClose]
  );

  return (
    <Modal open={open} onClose={onClose}>
      <Anime open={open} duration={1000}>
        <Holder role="ul">
          {(currencyData.payload || []).map(currency => (
            <ListItem key={currency.code} onClick={handleClick(currency)} role="li">
              <CurrencyCode>{currency.code}</CurrencyCode>
              <CurrencySubText>{currency.name}</CurrencySubText>
            </ListItem>
          ))}
        </Holder>
      </Anime>
    </Modal>
  );
};

export default CurrencyChooser;
