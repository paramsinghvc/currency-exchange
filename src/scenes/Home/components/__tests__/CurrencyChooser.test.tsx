import React from "react";
import CurrencyChooser from "../CurrencyChooser";
import renderer from "react-test-renderer";

describe("Should render CurrencyChooser properly", () => {
  it("should return null open is false", () => {
    const currencyChooser = renderer.create(
      <CurrencyChooser
        open={false}
        onClose={() => {}}
        onChange={() => {}}
        currencyData={{ payload: [], failure: false, pending: false, success: true, errors: null }}
      />
    );
    expect(currencyChooser.toJSON()).toMatchSnapshot();
  });
});
