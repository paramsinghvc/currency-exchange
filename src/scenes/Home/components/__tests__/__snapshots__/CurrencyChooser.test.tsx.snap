// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CurrencyChooser properly should return null open is false 1`] = `null`;
